frontend/app/src/App.test.tsx

```diff
@@ -1735,4 +1736,19 @@ describe("handles HostCommunication messaging", () => {
 
     expect(wrapper.find("DeployButton")).toHaveLength(0)
   })
+
+  it("does not relay custom parent messages by default", () => {
```

Comment:
nit: it would be good to add one more test to check that it is actually sent out if the setting is activated, e.g. by changing the app config state on the instance and mocking `sendMessageToHost`. An e2e test for this would be even better, but I think e2e for this might be quite complicated.